apps/imb/reframe_imb.py

```diff
@@ -0,0 +1,119 @@
+""" Performance tests using Intel MPI Benchmarks.
+
+    See README.md for details.
+
+    For parallel-transfer tests (i.e. uniband and biband) which use multiple process pairs, the -npmin flag is passed so that only the specified number of processes is run on each test.
+    This is the easiest way of ensuring proper process placement on the two nodes.
+
+    Run using e.g.:
+
+        reframe/bin/reframe -C reframe_config.py -c apps/imb/ --run --performance-report
+"""
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+import sys, os, math
+from collections import namedtuple
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..'))
+from modules.imb import read_imb_out
+from modules.reframe_extras import ScalingTest
+from modules.utils import identify_build_environment
+
+Metric = namedtuple('Metric', ['column', 'function', 'unit', 'label'])
+
+class IMB_base(rfm.RegressionTest):
+    METRICS = []
+
+    def __init__(self):
+        self.valid_systems = ['*:compute-node']
+        self.valid_prog_environs = ['intel']
```

Comment:
Related to the comment in `reframe_config.py`: I think this doesn't have any effect when the Spack build system is used, does it?  If you want to force Intel compilers you should use below a Spack spec like `intel-mpi-benchmarks@2019.6%intel`, and `intel-mpi-benchmarks@2019.6%intel^intel-mpi` if you also want to use Intel MPI.

Reply:
Ok, I thought this programming environment would get somehow passed down to spack, but it seems to run happily in the `gnu` environment as well.